iceoryx2/src/service/port_factory/event.rs

```diff
@@ -127,3 +130,53 @@ impl<'config, Service: service::Details<'config>> PortFactory<'config, Service>
         PortFactoryListener { factory: self }
     }
 }
+
+impl<'config> fmt::Display for PortFactory<'config, zero_copy::Service<'config>> {
```

Comment:
I think it would be better to introduce the second generic parameter here: `Service: service::Details<'config>` and then replace the line `f.debug_struct("zero_copy::Service")` (136) with

```rust
f.debug_struct(format!("{}", core::any::type_name::<Service>()))
```

With this we reduce the code duplication and when we introduce another service variant we do not have to add here a third version (which may is even forgotten).

@hydroid7 What do you think?

Reply:
Oh, `core::any::type_name` is a game changer here! :tada: